Documentation/git-submodule.txt

```diff
@@ -358,6 +358,16 @@ the submodule itself.
 	If the key `submodule.$name.update` is set to `rebase`, this option is
 	implicit.
 
+--reset-hard::
+	This option is only valid for the update command.
+	Hard reset current state to the commit recorded in the	superproject.
+    If this option is given, the submodule's HEAD will not get detached
+    if it was not detached before. Note that, like with a regular
+    git reset --hard  no safe-guards are in place to prevent jumping
```

Comment:
Could you quote `git reset --hard` such that it is highlighted? (Also double space after that command?)

Reply:
thanks, fixed

Also added to this PR an additional improvement (IMHO) to the tests, will follow up on the mailing list now